corehq/apps/hqwebapp/management/commands/copy_required_static_files.py

```diff
@@ -0,0 +1,133 @@
+            self.stderr.write("STATIC_ROOT is not configured")
+            return
+
+        REQUIRED_STATIC_FILES_DIR = 'REQUIRED_STATIC_FILES'
+        # Create STATIC_FILTERED directory in project root
+        static_filtered = os.path.join(settings.BASE_DIR, REQUIRED_STATIC_FILES_DIR)
+
+        # Create directory if it doesn't exist, or clean it if it does
+        if os.path.exists(static_filtered):
+            shutil.rmtree(static_filtered)
+        os.makedirs(static_filtered)
+
+        def get_app_name(app_path):
+            """Extract the final app name from a Python path."""
+            # Handle both string paths and config objects
+            if hasattr(app_path, 'name'):
+                app_path = app_path.name
+            # Split on dots and get the last part
+            parts = app_path.split('.')
+
+            # Handle special cases like 'django.contrib.admin' -> 'admin'
+            if 'contrib' in parts:
+                return parts[-1]
+
+            # Handle cases like 'corehq.apps.users' -> 'users'
+            if 'apps' in parts:
+                return parts[-1]
+
+            # Default to last part of path
+            return parts[-1]
+
+        # Get list of installed app paths
+        installed_apps = {get_app_name(app.name): app.path for app in apps.get_app_configs()}
+        print(installed_apps)
+        # Additional directories to copy
+        additional_dirs = ['jsi18n', 'webpack', 'webpack_b3', 'CACHE']
+
+        files_to_copy = ['build.b3.js', 'build.b3.txt', 'build.b5.js', 'build.b5.txt', 'build.js', 'build.txt']
+
+        def should_copy_directory(dir_name):
+            # Skip node_modules
+            if 'node_modules' in dir_name:
+                return False
+
+            # Check if directory is an installed app
+            if dir_name in installed_apps:
+                return True
+
+            # Check if directory is in additional_dirs
+            if dir_name in additional_dirs:
+                return True
+
+            return False
+
+        def copy_directory(src, dst):
+            if not os.path.exists(dst):
+                os.makedirs(dst)
+
+            for item in os.listdir(src):
+                s = os.path.join(src, item)
+                d = os.path.join(dst, item)
+
+                if os.path.isdir(s):
+                    if 'node_modules' not in s:
+                        copy_directory(s, d)
+                else:
+                    if not os.path.exists(os.path.dirname(d)):
+                        os.makedirs(os.path.dirname(d))
+                    shutil.copy2(s, d)
+
+        # Copy files
+        copied_count = 0
+        for item in os.listdir(static_root):
+            src_path = os.path.join(static_root, item)
+            dst_path = os.path.join(static_filtered, item)
+
+            if os.path.isdir(src_path) and should_copy_directory(item):
+                self.stdout.write(f"Copying {item}...")
+                copy_directory(src_path, dst_path)
```

Comment:
Can `copy_directory` be replaced with [`shutil.copytree`](https://docs.python.org/3.8/library/shutil.html?highlight=shutil#shutil.copytree)?
```suggestion
                shutil.copytree(src_path, dst_path, dirs_exist_ok=True)
```

Could also pass [`ignore=ignore_patterns('node_modules')`](https://docs.python.org/3.8/library/shutil.html?highlight=shutil#shutil.ignore_patterns) if these directories may contain `node_modules` that should be excluded. Is that necessary?

Reply:
I don't think these directories would have `node_modules` but we can add it to ignore list 🤔

Reply:
https://github.com/dimagi/commcare-hq/pull/35427/commits/5d645896d21306917889c6b277d3bd6b4f7dae01